## 1. Inspecting the data
<p>In this notebook, we will take a look at data on SATs across public schools in New York City. Our database contains a single table:</p>
<h1 id="schools"><code>schools</code></h1>
<table>
<thead>
<tr>
<th>column</th>
<th>type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>school_name</code></td>
<td><code>varchar</code></td>
<td>Name of school</td>
</tr>
<tr>
<td><code>borough</code></td>
<td><code>varchar</code></td>
<td>Borough that the school is located in</td>
</tr>
<tr>
<td><code>building_code</code></td>
<td><code>varchar</code></td>
<td>Code for the building</td>
</tr>
<tr>
<td><code>average_math</code></td>
<td><code>int</code></td>
<td>Average math score for SATs</td>
</tr>
<tr>
<td><code>average_reading</code></td>
<td><code>int</code></td>
<td>Average reading score for SATs</td>
</tr>
<tr>
<td><code>average_writing</code></td>
<td><code>int</code></td>
<td>Average writing score for SATs</td>
</tr>
<tr>
<td><code>percent_tested</code></td>
<td><code>numeric</code></td>
<td>Percentage of students completing SATs</td>
</tr>
</tbody>
</table>

In [128]:
%%sql
postgresql:///schools

select * 
from schools
limit 10;

10 rows affected.


[('New Explorations into Science, Technology and Math High School', 'Manhattan', 'M022', 657, 601, 601, None),
 ('Essex Street Academy', 'Manhattan', 'M445', 395, 411, 387, 78.9),
 ('Lower Manhattan Arts Academy', 'Manhattan', 'M445', 418, 428, 415, 65.1),
 ('High School for Dual Language and Asian Studies', 'Manhattan', 'M445', 613, 453, 463, 95.9),
 ('Henry Street School for International Studies', 'Manhattan', 'M056', 410, 406, 381, 59.7),
 ('Bard High School Early College', 'Manhattan', 'M097', 634, 641, 639, 70.8),
 ('Urban Assembly Academy of Government and Law', 'Manhattan', 'M445', 389, 395, 381, 80.8),
 ('Marta Valle High School', 'Manhattan', 'M025', 438, 413, 394, 35.6),
 ('University Neighborhood High School', 'Manhattan', 'M446', 437, 355, 352, 69.9),
 ('New Design High School', 'Manhattan', 'M445', 381, 396, 372, 73.7)]

## 2. Finding missing values

In [130]:
%%sql

SELECT COUNT(percent_tested) as num_tested_missing, COUNT(*) as num_schools
from schools;

 * postgresql:///schools
1 rows affected.


[(20, 375)]

## 3. Schools by building code

In [132]:
%%sql

select count(distinct building_code) as num_school_buildings
from schools

 * postgresql:///schools
1 rows affected.


[(233,)]

## 4. Best schools for math
<p>Out of 375 schools, only 233 (62%) have a unique <code>building_code</code> </p>

In [134]:
%%sql

-- Filter for average_math 640 or higher

select school_name, average_math
from schools
where average_math > 640
order by average_math desc;

 * postgresql:///schools
10 rows affected.


[('Stuyvesant High School', 754),
 ('Bronx High School of Science', 714),
 ('Staten Island Technical High School', 711),
 ('Queens High School for the Sciences at York College', 701),
 ('High School for Mathematics, Science, and Engineering at City College', 683),
 ('Brooklyn Technical High School', 682),
 ('Townsend Harris High School', 680),
 ('High School of American Studies at Lehman College', 669),
 ('New Explorations into Science, Technology and Math High School', 657),
 ('Eleanor Roosevelt High School', 641)]

## 5. Lowest reading score


In [136]:
%%sql

select min(average_reading) as lowest_reading
from schools

 * postgresql:///schools
1 rows affected.


[(302,)]

## 6. Best writing school

In [138]:
%%sql

select school_name, max(average_writing)  as max_writing
from schools
group by school_name
order by 2 desc
limit 1

 * postgresql:///schools
1 rows affected.


[('Stuyvesant High School', 693)]

## 7. Top 10 schools


In [140]:
%%sql

select school_name, (average_math+average_writing+average_reading) as average_sat
from schools
group by school_name
order by 2 desc
limit 10

 * postgresql:///schools
10 rows affected.


[('Stuyvesant High School', 2144),
 ('Staten Island Technical High School', 2041),
 ('Bronx High School of Science', 2041),
 ('High School of American Studies at Lehman College', 2013),
 ('Townsend Harris High School', 1981),
 ('Queens High School for the Sciences at York College', 1947),
 ('Bard High School Early College', 1914),
 ('Brooklyn Technical High School', 1896),
 ('Eleanor Roosevelt High School', 1889),
 ('High School for Mathematics, Science, and Engineering at City College', 1889)]

## 8. Ranking boroughs

In [142]:
%%sql

select borough, count(school_name) as num_schools, sum(average_math+average_writing+average_reading)/count(school_name) as average_borough_sat
from schools 
group by borough
order by 3 desc
                                                                                                        
                                                                                                        


 * postgresql:///schools
5 rows affected.


[('Staten Island', 10, 1439),
 ('Queens', 69, 1345),
 ('Manhattan', 89, 1340),
 ('Brooklyn', 109, 1230),
 ('Bronx', 98, 1202)]

## 9. Brooklyn numbers
<p>It appears that schools in Staten Island, on average, produce higher scores across all three categories. </p>


In [144]:
%%sql

select school_name, average_math
from schools
where borough = 'Brooklyn'
order by  2 desc
limit 5

 * postgresql:///schools
5 rows affected.


[('Brooklyn Technical High School', 682),
 ('Brooklyn Latin School', 625),
 ('Leon M. Goldstein High School for the Sciences', 563),
 ('Millennium Brooklyn High School', 553),
 ('Midwood High School', 550)]